go.mod

```diff
@@ -30,10 +30,14 @@ require (
 	github.com/spf13/cobra v0.0.5 // indirect
 	github.com/stretchr/testify v1.4.0
 	github.com/tmc/grpc-websocket-proxy v0.0.0-20200122045848-3419fae592fc
+	golang.org/x/net v0.0.0-20200301022130-244492dfa37a
+	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
+	google.golang.org/api v0.20.0
 	google.golang.org/genproto v0.0.0-20200430143042-b979b6f78d84
 	google.golang.org/grpc v1.28.0
 	google.golang.org/protobuf v1.22.0
 	gopkg.in/yaml.v2 v2.2.8
+	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
```

Comment:
Why do we need this too? Now we have are using 4 different libraries. Can you use `"sigs.k8s.io/yaml"` instead? I'd like to eventually refactor everything to use that.

Reply:
Going to create a separate ticket to investigate this.

- We use yaml.v3 to support reading (and writing) YAML with comments, inline comments, and other metadata.
- We use a pkg/config_types.go in CLI to marshal / unmarshal `params.yaml`
- Same package is used in core to dynamically generate the s3 access and secret keys, for Workflows

I think, ideally, we should use yaml.v3

Reply:
https://github.com/onepanelio/core/issues/409